### PR TITLE
Add changes to the Krell packages with version updates and ranges for  the latest release.

### PR DESCRIPTION
**Pull request by jgalarowicz**
Add changes to the Krell packages with version updates and ranges for  the latest release.   Change the ranges on versions, so the packages don't have to change as much when a new release is made.  Also, the changes allows the development version of OpenSpeedShop to use the develop version of Dyninst.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py [hunk 1]
```diff
@@ -51,50 +51,44 @@ class CbtfArgonavisGui(QMakePackage):
     url = "https://github.com/OpenSpeedShop/cbtf-argonavis-gui.git"
 
     version('1.3.0.0', branch='1.3.0.0',
-            git='https://github.com/OpenSpeedShop/cbtf-argonavis-gui.git')
+            git='https://github.com/OpenSpeedShop/cbtf-argonavis-gui.git', preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py]: This change shouldn't be necessary, Spack should already be choosing 1.3.0.0 by default.

File: var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py [hunk 2]
```diff
@@ -51,50 +51,44 @@ class CbtfArgonavisGui(QMakePackage):
...
-
-    # To specify ^elfutils@0.170 on the command line spack
-    # apparently needs/wants this dependency explicity here
-    # even though it is referenced downstream
-    depends_on("elf", type="link")
+    depends_on("cmake@3.0.2:", type='build')
 
     depends_on('qt@4.8.6:', when='@develop')
-    depends_on('qt@5.10.0', when='@1.3.0.0:')
+    depends_on('qt@5.10.0', when='@1.3.0.0:9999')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py]: I've always been confused by the dependencies in this package. It seems like the develop version supports older versions of dependencies than the latest release? It's usually the other way around...

File: var/spack/repos/builtin/packages/openspeedshop/package.py [hunk 3]
```diff
@@ -67,7 +67,7 @@ class Openspeedshop(CMakePackage):
 
     # Use when the git repository is available
     version('2.3.1.4', branch='2.3.1.4',
-            git='https://github.com/OpenSpeedShop/openspeedshop.git')
+            git='https://github.com/OpenSpeedShop/openspeedshop.git', preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openspeedshop/package.py]: This should already be the default version, no need for this change.

Comment (jgalarowicz) [reply var/spack/repos/builtin/packages/openspeedshop/package.py]: Same as above:
I think it was picking 2.3.1.4 as you say without the preferred, but spack was not following the same path as if the preferred statement clause is present.  The build was getting concretization errors without the preferred clause.

File: var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py (hunk 2 above)
Comment (jgalarowicz) [reply var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py]: We wanted to go with a proven path for the release versions and let spack pick the latest version available for develop.  I could make the develop version start at a higher level qt version.

File: var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py (hunk 1 above)
Comment (jgalarowicz) [reply var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py]: I think it was picking 1.3.0.0 as you say without the preferred, but spack was not following the same path as if the preferred statement clause is present.  The build was getting concretization errors without the preferred clause.

File: var/spack/repos/builtin/packages/openspeedshop/package.py (hunk 3 above)
Comment (jgalarowicz) [reply var/spack/repos/builtin/packages/openspeedshop/package.py]: @adamjstewart I will retest this and the cbtf-argonavis-gui one too.

File: var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py (hunk 2 above)
Comment (jgalarowicz) [reply var/spack/repos/builtin/packages/cbtf-argonavis-gui/package.py]: @adamjstewart I will update the qt develop version asked for.

**Comment by jgalarowicz** [comment]
@adamjstewart I updated the files to reflect your comments and suggestions.  Thanks!

**Pull request merged**